content/src/ports/rateLimitDeploymentCacheMap.ts

```diff
@@ -0,0 +1,115 @@
+import { ILoggerComponent } from '@well-known-components/interfaces'
+import { EntityType, Pointer, Timestamp } from 'dcl-catalyst-commons'
+import ms from 'ms'
+import NodeCache from 'node-cache'
+import { AppComponents } from '../types'
+
+export type IRateLimitDeploymentCacheMapComponent = {
+  newDeployment(entityType: EntityType, pointers: Pointer[], localTimestamp: Timestamp): void
+  isRateLimited(entityType: EntityType, pointers: Pointer[]): boolean
+}
+
+export type RateLimitConfig = {
+  defaultTtl: number
+  defaultMax: number
+  entitiesConfigTtl: Map<EntityType, number>
+  entitiesConfigMax: Map<EntityType, number>
+}
+
+export function createRateLimitDeploymentCacheMap(
+  components: Pick<AppComponents, 'logs'>,
+  rateLimitConfig: RateLimitConfig
+): IRateLimitDeploymentCacheMapComponent {
+  const logs: ILoggerComponent.ILogger = components.logs.getLogger('RateLimitDeploymentCacheMapComponent')
+
+  const deploymentCacheMap: Map<EntityType, { cache: NodeCache; maxSize: number }> = generateDeploymentCacheMap(
+    logs,
+    rateLimitConfig
+  )
+
+  function getFromCache(entityType: EntityType): { cache: NodeCache; maxSize: number } {
+    const cache = deploymentCacheMap.get(entityType)
+    if (!cache) {
+      throw new Error(`Invalid Entity Type: ${entityType}`)
+    }
+    return cache
+  }
+
+  return {
+    newDeployment(entityType: EntityType, pointers: Pointer[], localTimestamp: Timestamp): void {
+      const cacheByEntityType = getFromCache(entityType)
+      for (const pointer in pointers) {
+        cacheByEntityType.cache.set(pointer, localTimestamp)
+      }
+    },
+
+    /** Check if the entity should be rate limit: no deployment has been made for the same pointer in the last ttl
+     * and no more than max size of deployments were made either   */
+    isRateLimited(entityType: EntityType, pointers: Pointer[]): boolean {
+      const cacheByEntityType = getFromCache(entityType)
+      return (
+        pointers.some((p) => !!cacheByEntityType.cache.get(p)) ||
+        cacheByEntityType.cache.stats.keys > cacheByEntityType.maxSize
+      )
+    }
+  }
+}
+
+function generateDeploymentCacheMap(
+  logs: ILoggerComponent.ILogger,
+  rateLimitConfig: RateLimitConfig
+): Map<EntityType, { cache: NodeCache; maxSize: number }> {
+  const envMaxPerEntity: Map<EntityType, number> = getMaxPerEntityMapConfig(rateLimitConfig.entitiesConfigMax)
+  const envTtlPerEntity: Map<EntityType, number> = getTtlPerEntityMapConfig(rateLimitConfig.entitiesConfigTtl)
```

Comment:
Could those two functions be merged into one function returning a `Record<EntityName, { max: number, ttl: number }>` instead of two maps?

Reply:
That really makes sense! Will do it